blockchain/block.go

```diff
@@ -48,10 +50,18 @@ func (bh *BlockHeader) Marshal() []byte {
 	buf = append(buf, bh.Target.Marshal()...)
 	buf = append(buf, tempBufTime...)
 	buf = append(buf, tempBufNonce...)
+	buf = append(buf, bh.ExtraData...)
 
 	return buf
 }
 
+// Len returns the length in bytes of the BlockHeader.
+func (bh *BlockHeader) Len() int {
+	l := 2*(32/8) + 64/8 + 2*HashLen
```

Comment:
been looking in to this, unsafe.Sizeof causes some issues because in the ExtraData byte array Sizeof returns the size of the slice descriptor, not the size of the memory referenced by the slice.